prometheus/go_collector_go117.go

```diff
@@ -0,0 +1,316 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.17
+// +build go1.17
+
+package prometheus
+
+import (
+	"math"
+	"runtime"
+	"runtime/metrics"
+	"strings"
+
+	"github.com/golang/protobuf/proto"
```

Comment:
The CI is not happy about the import. This is not related to this PR, it can be handled separately.

Reply:
Add the following line prior to this one:
```Go
//nolint:staticcheck // Ignore SA1019. Need to keep deprecated package for compatibility.
```

Reply:
added the lint, if I did it right I'll resolve this.